src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/CompiledXPathExpr.cs

```diff
@@ -51,14 +52,14 @@ public override void AddSort(object expr, IComparer comparer)
             // sort makes sense only when we are dealing with a query that
             // returns a nodeset.
             Query evalExpr;
-            string query = expr as string;
+            string? query = expr as string;
```

Comment:
double check if comparer is non-null, I've seen many APIs where comparer argument accepts null

Reply:
This is only used by `AddSort` method on line 80 which passes a new `XPathComparerHelper` so we are fine.